client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,15 @@
+import { Route, RouteProps, Redirect } from 'react-router-dom';
+import { User } from '../../interface/User';
+
+interface Props extends RouteProps {
+  loggedInUser: User | null | undefined;
+}
+
+const ProtectedRoute = ({ loggedInUser, ...routeProps }: Props) => {
+  if (loggedInUser) {
```

Comment:
This lives in the context, so no need to pass it down - we can simply use the context in this component too:
```
const { loggedInUser } = useAuth();
```